internal/config/profile.go

```diff
@@ -258,40 +260,71 @@ func (p *Profile) SetPrivateAPIKey(v string) {
 	p.Set(privateAPIKey, v)
 }
 
-// AuthToken get configured public api key.
-func AuthToken() string { return p.AuthToken() }
-func (p *Profile) AuthToken() string {
-	return p.GetString(authToken)
+// AccessToken get configured access token.
+func AccessToken() string { return p.AccessToken() }
+func (p *Profile) AccessToken() string {
+	return p.GetString(accessToken)
 }
 
-// Access will return a jwt.Token and jwt.RegisteredClaims.
-// This method won't verify the token signature, it's only safe to use to get the token claims.
-func Access() (*jwt.Token, jwt.RegisteredClaims, error) { return p.Access() }
-func (p *Profile) Access() (*jwt.Token, jwt.RegisteredClaims, error) {
-	c := jwt.RegisteredClaims{}
-	// ParseUnverified is ok here, only want to make sure is a JWT and to get the claims for a Subject
-	token, _, err := new(jwt.Parser).ParseUnverified(p.AuthToken(), &c)
-	return token, c, err
+// SetAccessToken set configured access token.
+func SetAccessToken(v string) { p.SetAccessToken(v) }
+func (p *Profile) SetAccessToken(v string) {
+	p.Set(accessToken, v)
 }
 
-// SetAuthToken set configured publicAPIKey.
-func SetAuthToken(v string) { p.SetAuthToken(v) }
-func (p *Profile) SetAuthToken(v string) {
-	p.Set(authToken, v)
-}
-
-// RefreshToken get configured private api key.
+// RefreshToken get configured refresh token.
 func RefreshToken() string { return p.RefreshToken() }
 func (p *Profile) RefreshToken() string {
 	return p.GetString(refreshToken)
 }
 
-// SetRefreshToken set configured private api key.
+// SetRefreshToken set configured refresh token.
 func SetRefreshToken(v string) { p.SetRefreshToken(v) }
 func (p *Profile) SetRefreshToken(v string) {
 	p.Set(refreshToken, v)
 }
 
+// Token gets configured auth.Token.
+func Token() (*auth.Token, error) { return p.Token() }
+func (p *Profile) Token() (*auth.Token, error) {
+	if p.AccessToken() == "" || p.RefreshToken() == "" {
+		return nil, nil
+	}
+	c, err := p.tokenClaims()
+	if err != nil {
+		return nil, err
+	}
+	var e time.Time
+	if c.ExpiresAt != nil {
+		e = c.ExpiresAt.Time
+	}
+	t := &auth.Token{
+		AccessToken:  p.AccessToken(),
+		RefreshToken: p.RefreshToken(),
+		TokenType:    "Bearer",
+		Expiry:       e,
+	}
+	return t, nil
+}
+
+// AccessTokenSubject will return the encoded subject in a JWT.
+// This method won't verify the token signature, it's only safe to use to get the token claims.
```

Comment:
[q] where and how do we verify the token?

Reply:
Server side on use